dotnet/src/Connectors/Connectors.Google/Core/Gemini/GeminiFunctionToolResult.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.Connectors.Google;
+
+/// <summary>
+/// Represents the result of a Gemini function tool call.
+/// </summary>
+public sealed class GeminiFunctionToolResult
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GeminiFunctionToolResult"/> class.
+    /// </summary>
+    /// <param name="toolCall">The called function.</param>
+    /// <param name="functionResult">The result of the function.</param>
+    public GeminiFunctionToolResult(GeminiFunctionToolCall toolCall, FunctionResult functionResult)
+    {
+        Verify.NotNull(toolCall);
+        Verify.NotNull(functionResult);
+
+        this.FunctionResult = functionResult;
+        this.FullyQualifiedName = toolCall.FullyQualifiedName;
+    }
+
+    /// <summary>
+    /// Gets the result of the function.
+    /// </summary>
+    public FunctionResult FunctionResult { get; }
```

Comment:
Having a dependency on the non-serializable FunctionResult class makes the GeminiFunctionToolResult non-serializable as well. As a result, it will be impossible to serialize chat history containing function call results.

Reply:
So maybe we can do FunctionResult serializable?

Reply:
We can try, but after the exercise, most likely, we will see that the Function and ValueType properties are not serializable/deserializable and need to be ignored. This leaves us with not too many useful properties of the function result class.  
   
There could also be additional changes after those properties are ignored - they might need to be marked as nullable, and an additional constructor that does not have those properties in its signature needs to be added for deserialization purposes.  
   
We also need a way to pass exceptions to the connector in case the function invocation fails, and I don't see how function result class can help us with this.

Reply:
So I suggest leave it as is for now and update this when we finish tool call abstraction.

Reply:
Created #5658 for tracking in future.